stages/org.osbuild.oscap.remediation

```diff
@@ -128,15 +129,82 @@ def main(tree, options):
         cmd.extend(["--tailoring-file", tailoring])
 
+    if arf_results is not None:
+        # run with command in chroot so full path ok
+        cmd.extend(["--results-arf", f"{data_dir}/{arf_results}"])
+
+    cmd.extend(["--results", f"{data_dir}/{xccdf_results}"])
+
     cmd.append(datastream)
 
-    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False)
+    print("OpenSCAP (evaluate): ", repr(cmd))
+    # we need to inform the scanner that the system is offline (OSCAP_PROBE_ROOT="")
+    # and provide environment variables in OSCAP_CONTAINER_VARS as the probe
+    # does not have access to /proc
+    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False,
+                         env=dict(os.environ,
+                             OSCAP_PROBE_ROOT="",
+                             OSCAP_CONTAINER_VARS="container=bwrap-osbuild"))
 
     # oscap return values are:
     # 0 → success
-    # 2 → -- no error, but some checks/remediation failed
+    # 2 → no error, but some checks/remediation failed
+    if res.returncode not in (0, 2):
+        raise RuntimeError("oscap content evaluation failed")
+
+    # run openscap in chroot on the image tree (generate remediation script)
+    cmd = [
+        "/usr/sbin/chroot", tree,
+        "/usr/bin/oscap",
+        "xccdf", "generate"
+    ]
+
+    if verbose_level is not None:
+        cmd.extend(["--verbose", verbose_level])
+
+    if verbose_log is not None:
+        cmd.extend(["--verbose-log-file", f"{data_dir}/{verbose_log}.generate-fix"])
+
+    cmd.extend(["--profile", profile_id])
+    cmd.extend(["fix", "--fix-type", "bash",
+                "--output", f"{data_dir}/{REMEDIATION_SCRIPT}"])
+
+    cmd.append(f"{data_dir}/{xccdf_results}")
+
+    print("OpenSCAP (generate remediation): ", repr(cmd))
+    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False)
+
     if res.returncode not in (0, 2):
-        raise RuntimeError("oscap content evaluation and remediation failed")
+        raise RuntimeError("oscap failed to generate remediation script")
+
+    # run the remediation script
+    cmd = [
+        "/usr/sbin/chroot", tree,
+        "/usr/bin/bash",
+        f"{data_dir}/{REMEDIATION_SCRIPT}"
+    ]
+
+    log = open(f"{tree}/{data_dir}/{verbose_log}.remediation", "w") if verbose_log is not None else None
+
+    print("OpenSCAP remediation script cmd: ", repr(cmd))
+    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, stderr=log, check=False)
+
+    if log is not None:
+        log.close()
+
+    # pack result files, it could save dozens of megabytes
+    cmd = [
+        "/usr/sbin/chroot", tree,
+        "/usr/bin/xz"
+    ]
+
+    if arf_results is not None:
+        cmd.extend([f"{data_dir}/{arf_results}"])
+
+    cmd.extend([f"{data_dir}/{xccdf_results}"])
+
+    print("Pack the result files: ", repr(cmd))
+    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False)
 
```

Comment:
The default footprint in `data_dir` (`/root` if omitted) is:
- oscap_eval_xccdf_results.xml.xz (up to 10 Mb in the worst case)
- oscap_remediation.bash (about 1 Mb).

Reply:
@achilleas-k, is there a better way we could handle compressing the output?

Reply:
I would say that this is OK, but the approach has a few downsides...

The stage previously didn't depend on `xz` being available in the build environment. Therefore old manifest which don't have it installed in the build pipeline, would start failing if they used this stage. For the sake of backwards compatibility, I would suggest to make the compression optional (based on the availability of `xz` or based on an explicit schema property).

Reply:
How exactly should I check for presence of `xz` in a canonical way if there is any?

Reply:
There is no standard way that we use AFAIK, so from the top of my head I would say:

1. Check that `/usr/bin/xz` exists before using it.
2. Gracefully handle the error from `subprocess.run()`.

However, thinking about this a bit more, I think that it may be better to make the compression an explicit option in the stage schema, instead of changing the stage behavior based on the build environment. This would make the stage behavior more predictable. The old behavior (not compressing anything) can be preserved by default, while the stage can fail in case one enables the compression in stage options and `xz` is not available.

Reply:
Well there was no 150 MB file stored in /root in old behavior. That's not nice for sure, I'd like this file to be compressed by default. Also, it is impossible to turn this option on from a blueprint so it won't be very convenient.

Reply:
> it is impossible to turn this option on from a blueprint

We can handle that internally in composer, so there would be no change to the blueprints

Reply:
Alright, explicit options it is.